### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by stefanalund** [EricssonResearch/openwebrtc#2]
Add support for Windows.

**Comment by Tonku** [comment]
I dont know anything about the build system and scripts... otherwise I could have put some time into this ...Btw I will put some effort on it in coming days..as I need windows support as well for a project I am working on...
Thanks for the input Guys
Kind Regards
Tonku

**Comment by agouaillard** [comment]
yes, you can convince me, I have not made my mind yet and i haven't started
coding. I won;t go ahead with something that is not consensual.
Actually i will be in stockholm on 26/27 of jan. shall we speak around a
coffee/beer?

On Fri, Jan 9, 2015 at 7:02 PM, Robert Swain <notifications@github.com>
wrote:

> I would answer basically the same as @sdroege <https://github.com/sdroege>.
> It would be a lot of work to write CMake build systems for all of the
> OpenWebRTC dependencies and then you would have to maintain them all.
>
> Our current plan is to move to cerbero soon which is what is used to build
> the GStreamer release binaries for all platforms (iOS, Android, Mac OS X,
> Linux, Windows.) We are in the testing phase of this move.
>
> Once the migration to cerbero is complete we will probably deprecate the
> current build scripts.
>
> All the packages needed to build GStreamer for Windows do currently build
> as part of cerbero. It would then, as mentioned above, not be too much work
> to get the extra dependencies and OpenWebRTC building and running on
> Windows.
>
> We then intend to maintain our additional recipes that we need for
> OpenWebRTC and we will also work with the GStreamer project to move recipes
> or modifications to cerbero there as desirable and necessary.
>
> Can we convince you to instead work with us on getting Windows support
> working using cerbero as the build system?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EricssonResearch/openwebrtc/issues/2#issuecomment-69320790>
> .
>



-- 
Alex. Gouaillard, PhD, PhD, MBA
------------------------------------------------------------------------------------
CTO - Temasys Communications, S'pore / Mountain View
President - CoSMo Software, Cambridge, MA
------------------------------------------------------------------------------------
sg.linkedin.com/agouaillard

   -

**Comment by stefanalund** [comment]
Sounds like a good idea, I'll send you an email and we can keep this issue clean :-)

**Comment by sdroege** [comment]
Any progress/plans here?

**Comment by hlevring** [comment]
@agouaillard yes it would definitely be nice to know if you made any progress. This is an awesome project but windows support is really a big missing feature.

**Comment by nirbheek** [comment]
Somewhat relatedly, #222 tracks our move to using Cerbero as the build system, which has support for building for Windows (cross-compiling) and on Windows (through mingw32, afaik, but maybe more). The GStreamer binaries for Windows are built using Cerbero, so most of the dependencies will work just fine there. If someone is interested in that, one can start there. :smile:

**Comment by sdroege** [comment]
The main work will probably be in getting gobject-introspection, javascriptcore and seed to work properly on Windows. Otherwise everything else should be more or less working already.

**Comment by superdump** [comment]
So basically the OpenWebRTC C API may work on windows already...

**Comment by hlevring** [comment]
We are preparing the design and development work on a new product where we want  to use OpenWebRTC. Plan is to start sometime in early April, so this seems like perfect timing. Thanks guys

**Comment by superdump** [comment]
For reference: https://github.com/EricssonResearch/openwebrtc/wiki/Building-OpenWebRTC-with-Cerbero

We have not yet tested anything on Windows though. I'll see about doing that at some point soon.

@sdroege what is the current status of GStreamer camera / audio sources and sinks on Windows?

**Comment by sdroege** [comment]
The C API should already work, yes... except for the missing device listing on Windows.

Audio source would be directsoundsrc (or wasapisrc but that is not great yet), sink directsoundsink (or wasapisink but see before). Video sink would be d3dvideosink or glimagesink. And video source, there's something for DirectShow (but deprecated by MS since years) and ksvideosrc (also deprecated now IIRC). But as AFAIK everything still comes with DirectShow drivers anyway.

**Comment by hlevring** [comment]
So from what I understand on the above then Windows should theoretically work out of the box in our use case. We start our project in the beginning of April and on the windows platform we only need to connect, receive and decode a webrtc video/audio feed which will be consumed in a directshow based application. If you know of any roadblock currently for this use case then please do let us know ?. We will make sure to update you in case we hit any issues

**Comment by stefanalund** [comment]
@hlevring we have now moved to the new build system (cerbero) as discussed above. Would be cool if you guys started looking in to getting things to compile on Windows as well. We will be here to support you :)

https://github.com/EricssonResearch/cerbero

**Comment by hlevring** [comment]
Ok we will do that late next week. Will keep you informed, on progress.

**Comment by nirbheek** [comment]
@hlevring Don't hesitate to come onto #openwebrtc on FreeNode IRC and message us.

**Comment by agouaillard** [comment]
Hi all,

We have been swamped until a week or so ago. The plan was indeed to wait
for cerebro to be in place before we do anything (as opposed to moving to
another build system to better support windows, a.k.a. cmake).

I'm working on other openwebrtc projects right now, but shall take a look
next week and maybe start reporting issues and work on it. Do not wait on
me though ;-)

Alex.

On Fri, Mar 27, 2015 at 3:54 PM, Stefan Ålund <notifications@github.com>
wrote:

> @hlevring <https://github.com/hlevring> we have now moved to the new
> build system (cerbero) as discussed above. Would be cool if you guys
> started looking in to getting things to compile on Windows as well. We will
> be here to support you :)
>
> https://github.com/EricssonResearch/cerbero
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EricssonResearch/openwebrtc/issues/2#issuecomment-86872642>
> .
>



-- 
Alex. Gouaillard, PhD, PhD, MBA
------------------------------------------------------------------------------------
CTO - Temasys Communications, S'pore / Mountain View
President - CoSMo Software, Cambridge, MA
------------------------------------------------------------------------------------
sg.linkedin.com/agouaillard

   -

**Comment by dcposch** [comment]
Looks like there's recent work making cerbero work on Windows

Any chance we'll be able to run openwebrtc on all platforms soon?

( @feross and I are interested in using it for [Webtorrent](https://github.com/feross/webtorrent) )

**Comment by superdump** [comment]
The situation is the same. Cerbero builds GStreamer for Windows. We need to make it also build the few additional packages needed by OpenWebRTC and OpenWebRTC itself.

**Comment by dcposch** [comment]
Makes sense. Which additional packages? LMK if we can help!

**Comment by jonahzheng** [comment]
windows?
